internal/envoy/runtime.go

```diff
@@ -57,6 +58,31 @@ type Runtime struct {
 	shutdownHooks []func(context.Context) error
 }
 
+// String is only used in tests. It is slow, but helps when debugging CI failures
+func (r *Runtime) String() string {
+	var stdout, stderr string
+	if r.OutFile != nil {
+		if b, err := os.ReadFile(r.OutFile.Name()); err == nil {
+			stdout = string(b)
+		}
+	}
+
+	if r.ErrFile != nil {
+		if b, err := os.ReadFile(r.ErrFile.Name()); err == nil {
+			stderr = string(b)
+		}
+	}
+
+	exitStatus := -1
+	if r.cmd != nil && r.cmd.ProcessState != nil {
+		if ws, ok := r.cmd.ProcessState.Sys().(syscall.WaitStatus); ok {
+			exitStatus = ws.ExitStatus()
+		}
+	}
+
+	return fmt.Sprintf("{stdout: %s, stderr: %s, exitStatus: %d}", stdout, stderr, exitStatus)
```

Comment:
how does actual output look like from this String? Is that human-readable? 😄

Reply:
you tell me :) I think it is ok https://github.com/tetratelabs/func-e/pull/348/files#diff-0f025fafb42983e9b6848a9ba595f8f38e9cfaa80cd1a5b4c6bb8de7c82d1835R110

Reply:
no I mean what if stderr and stdout become too long by actualy Envoy? 😄

Reply:
oh now I see you comments ^^ this is only for tests!

Reply:
yeah I agree if real console out, this would be a bad idea